### PR TITLE
SANDBOX-681: add ban reason to NewBannedUser func

### PR DESCRIPTION
**Pull request by rsoaresd**
# Description
- add ban reason to `NewBannedUser` func

## Issue ticket number and link
[SANDBOX-681](https://issues.redhat.com/browse/SANDBOX-681)

Associated PRs:
- wa, pr 24
- https://github.com/kubesaw/ksctl/pull/74

### CODE REVIEW

File: pkg/banneduser/banneduser.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ import (
 )
 
 // NewBannedUser creates a bannedUser resource
-func NewBannedUser(userSignup *toolchainv1alpha1.UserSignup, bannedBy string) (*toolchainv1alpha1.BannedUser, error) {
+func NewBannedUser(userSignup *toolchainv1alpha1.UserSignup, bannedBy string, banReason string) (*toolchainv1alpha1.BannedUser, error) {
```
Comment (mfrancisc) [inline pkg/banneduser/banneduser.go:13]: just a small detail
```suggestion
func NewBannedUser(userSignup *toolchainv1alpha1.UserSignup, bannedBy, banReason string) (*toolchainv1alpha1.BannedUser, error) {
```

**Review by mfrancisc — APPROVED** [review]
Nice 👍

**Review by alexeykazakov — APPROVED** [review]
Looks good. 
But don't forget to check all the repos (host, member, e2e, reg-service, ksctl, sre) and update them if they use this API.

**Review by ranakan19 — APPROVED** [review]
looks good, nothing more to add to existing feedback from alexey and francisc

**Comment by MatousJobanek** [comment]
> But don't forget to check all the repos (host, member, e2e, reg-service, ksctl, sre) and update them if they use this API.

I think that we use this function in ksctl and workload-analyzer only

File: pkg/banneduser/banneduser.go (hunk 1 above)
Comment (rsoaresd) [reply pkg/banneduser/banneduser.go:13]: Thanks!!

**Comment by rsoaresd** [comment]
> > But don't forget to check all the repos (host, member, e2e, reg-service, ksctl, sre) and update them if they use this API.
> 
> I think that we use this function in ksctl and workload-analyzer only

Yes!

**Review by fbm3307 — APPROVED** [review]

**Pull request merged**